quickjs-libc.c

```diff
@@ -2089,6 +2102,9 @@ static JSValue js_os_clearTimeout(JSContext *ctx, JSValue this_val,
     if (!th)
         return JS_EXCEPTION;
     unlink_timer(JS_GetRuntime(ctx), th);
+    JS_FreeValue(ctx, th->func);
+    th->func = JS_UNDEFINED;
+    th->repeats = FALSE;
```

Comment:
This isn't strictly necessary, it just releases the callback's closure earlier than waiting for js_os_timer_finalizer to run.

Reply:
Makes sense!